### PR TITLE
Started adding logger standardization and exception handling

### PR DESCRIPTION
**Pull request by jpamarthi**
1. Bumped the version
2. Added new logger functionality to customize and control the logs based on the environment
3. Started adding exception handling to make the code robust. Tested with check_and_convert_date function.

### CODE REVIEW

File: src/pycatcher/catch.py [hunk 1]
```diff
@@ -14,31 +15,101 @@
 from scipy.special import inv_boxcox
 
 
-# Configure logging
-logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
-logger = logging.getLogger(__name__)
+# Configure logging based on environment with more detailed format
```
Comment (jpamarthi) [inline src/pycatcher/catch.py:18]: This function enables the log visibility as needed. The default level is WARNING if the user doesn't set it.

File: src/pycatcher/catch.py [hunk 2]
```diff
@@ -14,31 +15,101 @@
...
+        log.addHandler(handler)
+
+    return log
+
+
+# Initialize logger
+logger = setup_logger()
+
+
+class TimeSeriesError(Exception):
```
Comment (jpamarthi) [inline src/pycatcher/catch.py:49]: New class to handle TimeSeries error.

File: src/pycatcher/catch.py [hunk 3]
```diff
@@ -14,31 +15,101 @@
...
+    Attributes:
+        message (str): Explanation of the error
+    """
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class DataValidationError(Exception):
```
Comment (jpamarthi) [inline src/pycatcher/catch.py:65]: Class to handle data validation errors

File: src/pycatcher/catch.py [hunk 4]
```diff
@@ -14,31 +15,101 @@
...
+        raise DataValidationError("Input DataFrame cannot be None or empty")
 
-    # Check if the column is already in datetime format
-    if pd.api.types.is_datetime64_any_dtype(df[first_col_name]):
-        df[df.columns[0]] = df[df.columns[0]].apply(pd.to_datetime)
-        df = df.set_index(df.columns[0]).dropna()
-    else:
-        try:
-            # Attempt to convert the column to datetime
+    try:
```
Comment (jpamarthi) [inline src/pycatcher/catch.py:91]: Refactored the code to add logger and exception handling by leaving the existing functionality intact.

File: tests/conftest.py [hunk 5]
```diff
@@ -1,7 +1,14 @@
+import os
 import pytest
 from src.pycatcher import create_app
 from src.pycatcher.webapp import register_routes
 
+
+def pytest_configure(config):
```
Comment (jpamarthi) [inline tests/conftest.py:7]: Setting the logger level to CRITICAL to suppress the INFO messages while running test cases.

File: tests/test_catch.py [hunk 6]
```diff
@@ -4,10 +4,49 @@
...
-from src.pycatcher.catch import find_outliers_iqr, anomaly_mad, get_residuals, \
-    sum_of_squares, get_ssacf, detect_outliers_today_classic, detect_outliers_latest_classic, \
-    detect_outliers_classic, decompose_and_detect, detect_outliers_iqr
+from src.pycatcher.catch import (TimeSeriesError, DataValidationError, check_and_convert_date, find_outliers_iqr,
+    anomaly_mad, get_residuals, sum_of_squares, get_ssacf, detect_outliers_today_classic,
+    detect_outliers_latest_classic, detect_outliers_classic, decompose_and_detect, detect_outliers_iqr)
+
+
+@pytest.fixture
+def sample_df():
```
Comment (jpamarthi) [inline tests/test_catch.py:13]: Sample data to feed to test cases

File: tests/test_catch.py [hunk 7]
```diff
@@ -4,10 +4,49 @@
...
+    """Fixture for sample DataFrame with dates and values."""
+    return pd.DataFrame({
+        'date': pd.date_range(start='2022-01-01', periods=5),
+        'value': [10, 20, 30, 40, 50]
+    })
+
+
+class TestCheckAndConvertDate:
```
Comment (jpamarthi) [inline tests/test_catch.py:21]: New test case coverage for the function check_and_convert_date

**Comment by jpamarthi** [comment]
@aseemanand if this is looking good, I will finish adding this to all the functions this weekend to close out the logging and exception handling standards.

**Review by aseemanand — APPROVED** [review]
LGTM

**Pull request merged**